pkg/commands/artifact/run.go

```diff
@@ -233,6 +236,19 @@ func scan(ctx context.Context, opt Option, initializeScanner InitializeScanner,
 		}
 	}
 
+	// Do not load config file for secret scanning
+	if !slices.Contains(opt.SecurityChecks, types.SecurityCheckSecret) {
+		opt.SecretConfigPath = ""
+	} else {
+		ver := fmt.Sprintf("v%s", opt.AppVersion)
+		if opt.AppVersion == "dev" {
+			ver = "dev"
+		}
+		log.Logger.Info("Secret detection is enabled")
+		log.Logger.Info("If your scanning is slow, please try '--security-checks vuln' to disable secret scanning")
+		log.Logger.Infof("Please see also https://aquasecurity.github.io/trivy/%s/docs/secret/scanning/#recommendation for faster secret detection", ver)
+	}
```

Comment:
I don't like long `else` blocks, so I'd write next (it looks like another `if` blocks in this code):
```go
if slices.Contains(opt.SecurityChecks, types.SecurityCheckSecret) {
	log.Logger.Info("Secret detection is enabled")
	ver := fmt.Sprintf("v%s", opt.AppVersion)
	if opt.AppVersion == "dev" {
		ver = "dev"
	}
	log.Logger.Info("If your scanning is slow, please try '--security-checks vuln' to disable secret scanning")
	log.Logger.Infof("Please see also https://aquasecurity.github.io/trivy/%s/docs/secret/scanning/#recommendation for faster secret detection", ver)
} else {
        opt.SecretConfigPath = ""
}
```

Reply:
done https://github.com/aquasecurity/trivy/pull/2051/commits/2e277726fb33a1184ffd47a06ed7fe80bb797f3a